sdk/python/kfp/components/types/type_annotations.py

```diff
@@ -223,3 +229,14 @@ def is_artifact_class(artifact_class_or_instance: Type) -> bool:
     # we do not yet support non-pre-registered custom artifact types with instance_schema attribute
     return hasattr(artifact_class_or_instance, 'schema_title') and hasattr(
         artifact_class_or_instance, 'schema_version')
+
+
+def is_list_of_artifacts(
+    type_var: Union[Type[List[artifact_types.Artifact]],
+                    Type[artifact_types.Artifact]]
+) -> bool:
+    # the type annotation for this function's `type_var` parameter may not actually be a subclass of the KFP SDK's Artifact class for custom artifact types
+    return hasattr(
+        type_var, '__origin__'
+    ) and type_var.__origin__ == list and type_annotations.is_artifact_class(
```

Comment:
nit: can be shorten to `getattr(type_var, '__origin__', None) == list and ....`. Feel free to ignore.